src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -153,12 +202,18 @@ class DiscreteDistribution
    *    actually from this distribution.
    */
   void Train(const arma::mat& observations,
-             const arma::vec& probabilities);
+             const arma::mat& probabilities);
 
   //! Return the vector of probabilities.
-  const arma::vec& Probabilities() const { return probabilities; }
+  const arma::vec& Probabilities() const { return probabilities[0]; }
   //! Modify the vector of probabilities.
-  arma::vec& Probabilities() { return probabilities; }
+  arma::vec& Probabilities() { return probabilities[0]; }
+
+  //! Modify the vector Martix of probabilities.
+  const std::vector<arma::vec>& MultipleProbabilities() const { return probabilities; }
+
+  //! Return the vector Martix of probabilities.
+  std::vector<arma::vec>& MultipleProbabilities() { return probabilities; }
```

Comment:
I think `Probabilities()` is only used in the tests really so it's not a problem to make this change I think. Maybe a better solution is this:

```
const arma::vec& Probabilities(const size_t dim = 0)
{
  return probabilities[I];
}
```

That actually should be reverse compatible too which is nice!